app/src/main/kotlin/com/hippo/ehviewer/ui/main/RollingNumber.kt

```diff
@@ -0,0 +1,124 @@
+@file:Suppress("ktlint:standard:property-naming")
+
+package com.hippo.ehviewer.ui.main
+
+import androidx.compose.animation.core.animateIntSize
+import androidx.compose.animation.core.animateOffset
+import androidx.compose.animation.core.updateTransition
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.offset
+import androidx.compose.material3.LocalTextStyle
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clipToBounds
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.layout.layout
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.rememberTextMeasurer
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.IntSize
+import kotlin.math.PI
+import kotlin.math.atan2
+import kotlin.math.cos
+import kotlin.math.floor
+import kotlin.math.max
+import kotlin.math.sin
+
+@Composable
+fun RollingNumber(number: Int, style: TextStyle = LocalTextStyle.current) {
+    val string = remember(number) { "$number" }
+    val transition = updateTransition(string)
+    val textMeasurer = rememberTextMeasurer()
+    val size by transition.animateIntSize { str ->
+        remember(str) { textMeasurer.measure(str, style).size }
+    }
+    Row(
+        modifier = Modifier.clipToBounds().layout { measurable, constraints ->
+            val placeable = measurable.measure(constraints)
+            layout(width = placeable.width, height = size.height) {
+                placeable.place(0, 0)
+            }
+        },
+        horizontalArrangement = Arrangement.End,
+    ) {
+        val from = transition.currentState
+        val fromLen = from.length
+        val to = transition.targetState
+        val toLen = to.length
+        val max = max(fromLen, toLen)
+        repeat(max) { where ->
```

Comment:
10 -> 9 transition still feels wonky.
I suggest to use a fixed value here since the max value of `number` is known.

Reply:
Why it's known? number is mutable

Reply:
The number of total pages is known.

Reply:
> The number of total pages is known.

You are right, I will add a param

Reply:
Done